src/Bundle/DependencyInjection/PmgTwigWebpackExtension.php

```diff
@@ -45,20 +42,15 @@ public static function createWebpack(Packages $packages=null, $devMode, $baseUrl
      */
     protected function loadInternal(array $config, ContainerBuilder $container)
     {
+        $loader = new XmlFileLoader(
+            $container,
+            new FileLocator(__DIR__.'/../Resources/config')
+        );
+
         foreach ($config as $key => $val) {
             $container->setParameter(sprintf('pmg_twig_webpack.%s', $key), $val);
         }
 
-        $container->setDefinition('pmg_twig_webpack.webpack', new Definition(Webpack::class, [
-                new Reference('assets.packages', ContainerInterface::NULL_ON_INVALID_REFERENCE),
-                '%pmg_twig_webpack.dev_mode%',
-                '%pmg_twig_webpack.base_url%',
-            ]))
-            ->setFactory([__CLASS__, 'createWebpack']);
-
-        $container->setDefinition('pmg_twig_webpack.extension', new Definition(
-            WebpackExtension::class,
-            [new Reference('pmg_twig_webpack.webpack')]
-        ))->addTag('twig.extension');
+        $loader->load('services.xml');
```

Comment:
Why switch this to a file loader vs. in the extension itself?

Reply:
to use dependency injection `The "pmg_twig_webpack.webpack" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead.`